qiskit/test/mock/utils/fake_backend_builder.py

```diff
@@ -0,0 +1,331 @@
+        meas_lo_range = [[6.5, 7.5] for _ in range(self.n_qubits)]
+        u_channel_lo = [[UchannelLO(q=i, scale=1. + 0.j)] for i in range(len(self.cmap))]
+
+        return PulseBackendConfiguration(
+            backend_name=self.name,
+            backend_version=self.version,
+            n_qubits=self.n_qubits,
+            meas_levels=[0, 1, 2],
+            basis_gates=self.basis_gates,
+            simulator=False,
+            local=True,
+            conditional=True,
+            open_pulse=True,
+            memory=False,
+            max_shots=65536,
+            gates=[],
+            coupling_map=self.cmap,
+            n_registers=self.n_qubits,
+            n_uchannels=self.n_qubits,
+            u_channel_lo=u_channel_lo,
+            meas_level=[1, 2],
+            qubit_lo_range=qubit_lo_range,
+            meas_lo_range=meas_lo_range,
+            dt=1.3333,
+            dtm=10.5,
+            rep_times=[1000],
+            meas_map=meas_map,
+            channel_bandwidth=[],
+            meas_kernels=['kernel1'],
+            discriminators=['max_1Q_fidelity'],
+            acquisition_latency=[],
+            conditional_latency=[],
+            hamiltonian=hamiltonian
+        )
+
+    def build_defaults(self) -> PulseDefaults:
+        """Build backend defaults."""
+
+        qubit_freq_est = np.linspace(4.9, 5.1, self.n_qubits).tolist()
+        meas_freq_est = np.linspace(6.4, 6.6, self.n_qubits).tolist()
+        buffer = 10
+        pulse_library = [
+            {
+                'name': 'test_pulse_1',
+                'samples': [[0.0, 0.0], [0.0, 0.1]]
+            },
+            {
+                'name': 'test_pulse_2',
+                'samples': [[0.0, 0.0], [0.0, 0.1], [0.0, 1.0]]
+            },
+            {
+                'name': 'test_pulse_3',
+                'samples': [[0.0, 0.0], [0.0, 0.1], [0.0, 1.0], [0.5, 0.0]]
+            },
+            {
+                'name': 'test_pulse_4',
+                'samples': 7 * [[0.0, 0.0], [0.0, 0.1], [0.0, 1.0], [0.5, 0.0]]
+            }
+        ]
+
+        measure_command_sequence = [PulseQobjInstruction(name='acquire', duration=10, t0=0,
+                                                         qubits=list(range(self.n_qubits)),
+                                                         memory_slot=list(range(self.n_qubits))
+                                                         ).to_dict()]
+        measure_command_sequence += [PulseQobjInstruction(name='test_pulse_1',
+                                                          ch='m{}'.format(i), t0=0).to_dict()
+                                     for i in range(self.n_qubits)]
+
+        measure_command = Command.from_dict({
+            'name': 'measure',
+            'qubits': list(range(self.n_qubits)),
+            'sequence': measure_command_sequence
+        }).to_dict()
+
+        cmd_def = [measure_command]
+
+        for i in range(self.n_qubits):
```

Comment:
I think this should loop over `self.single_qubit_gates` and build a definition for each qubit, for each gate

Reply:
https://github.com/Qiskit/qiskit-terra/pull/3797/commits/a66464df5a1c1ac8eaff73dff1415cfcf71a9ebb